app/controllers/students_controller.rb

```diff
@@ -30,6 +29,11 @@ def show
     }
   end
 
+  def restricted_notes
+    raise Exceptions::EducatorNotAuthorized unless current_educator.can_view_restricted_notes
```

Comment:
I tested by changing `raise Exceptions::EducatorNotAuthorized unless current_educator.is_authorized_for_student(student)` to `raise Exceptions::EducatorNotAuthorized`. When I tried to visit `http://docker:3000/students/25/restricted_notes`, I was redirected to the NotAuthorized page.